ui/pages/confirmations/confirmation/useConfirmationRouting.ts

```diff
@@ -0,0 +1,25 @@
+import { useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { isEqual } from 'lodash';
+import { getUnapprovedTemplatedConfirmations } from '../../../selectors';
+
+/**
+ * A utility hook that parses an optional ID from the URL and
+ * returns the confirmation with that ID for routing purposes.
+ *
+ * @returns A pending confirmation if applicable.
+ */
+export function useConfirmationRouting() {
+  const pendingConfirmations = useSelector(
+    getUnapprovedTemplatedConfirmations,
+    isEqual,
+  );
+
+  const { id } = useParams<{ id: string }>();
```

Comment:
Minor, but as the ID is optional, should the type be `id?: string` or does `useParams` assume all params are optional?